algoliasearch/index.go

```diff
@@ -406,7 +406,11 @@ func (i *index) Move(name string) (UpdateTaskRes, error) {
 }
 
 func (i *index) MoveWithRequestOptions(name string, opts *RequestOptions) (UpdateTaskRes, error) {
-	return i.operation(name, "move", nil, opts)
+	res, err := i.operation(name, "move", nil, opts)
```

Comment:
Unfortunately this may be a breaking change. The index object no longer targets the old index, but the new one.

Shouldn't we consider move this to the v2 version?

Reply:
@aseure Unless we consider this a bug 😁

Reply:
Arghhh, you're right. Good catch. I'll base this PR on the `develop` branch then (next major release branch). Thank you.

Reply:
@nunomaduro At first, I wanted to consider it as a bug but you're right, it's a breaking change as one may have relied on this behaviour with this scenario for instance:

1. index := client.InitIndex("old_name")
2. index.AddObjects(/* pass 10 objects */)
3. index.Move("new_name")
4. index.AddObjects(/* pass 20 objects */)

In that case, "old_name" index contains the latest 20 objects and "new_name" contains the first 10 objects. However, with the proposed change, we would have all 30 objects on "new_name" index.

I've just made `develop` branch the new base branch of this PR so that it will only be included with the upcoming v3 of the client to avoid breaking the compatibility. Thanks for your thorough review.

Reply:
@aseure Nice, this is definitively this safest move. 👍